script/release-notes/index.js

```diff
@@ -1,473 +1,139 @@
-    if (latest === null) latest = b
-    if (semver.gt(semverify(b), semverify(latest))) {
-      latest = b
-    }
-  }
-  return latest
+const getTagsOnBranch = async (point) => {
+  const m = await getTagsOf('master')
+  if (point === 'master') return m
+  const masterTags = new Set(m)
+  return (await getTagsOf(point)).filter(t => !masterTags.has(t))
 }
 
-const commitBeforeTag = async (commit, tag) => {
-  const gitArgs = ['tag', '--contains', commit]
-  const tagDetails = await GitProcess.exec(gitArgs, gitDir)
-  if (tagDetails.exitCode === 0) {
-    return tagDetails.stdout.split('\n').includes(tag)
-  }
-  throw GitProcess.parseError(tagDetails.stderr)
+const getBranchOf = async (point) => {
+  const branches = (await runGit(['branch', '-a', '--contains', point]))
+    .split('\n')
+    .map(b => b.trim())
+    .filter(b => !!b)
+  const current = branches.find(b => b.startsWith('* '))
+  return current ? current.slice(2) : branches.shift()
 }
 
-const getCommitsMergedIntoCurrentBranchSincePoint = async (point) => {
-  return getCommitsBetween(point, 'HEAD')
+const getAllBranches = async () => {
+  return (await runGit(['branch', '--remote']))
+    .split('\n')
+    .map(b => b.trim())
+    .filter(b => !!b)
+    .filter(branch => branch !== 'origin/HEAD -> origin/master')
+    .sort()
 }
 
-const getCommitsBetween = async (point1, point2) => {
-  const gitArgs = ['rev-list', `${point1}..${point2}`]
-  const commitsDetails = await GitProcess.exec(gitArgs, gitDir)
-  if (commitsDetails.exitCode !== 0) {
-    throw GitProcess.parseError(commitsDetails.stderr)
-  }
-  return commitsDetails.stdout.trim().split('\n')
+const getStabilizationBranches = async () => {
+  return (await getAllBranches())
+    .filter(b => /^origin\/\d+-\d+-x$/.test(b))
 }
 
-const TITLE_PREFIX = 'Merged Pull Request: '
+const getPreviousStabilizationBranch = async (current) => {
+  const stabilizationBranches = (await getStabilizationBranches())
+    .filter(b => b !== current && b !== `origin/${current}`)
 
-const getCommitDetails = async (commitHash) => {
-  const commitInfo = await (await fetch(`https://github.com/electron/electron/branch_commits/${commitHash}`)).text()
-  const bits = commitInfo.split('</a>)')[0].split('>')
-  const prIdent = bits[bits.length - 1].trim()
-  if (!prIdent || commitInfo.indexOf('href="/electron/electron/pull') === -1) {
-    console.warn(`WARNING: Could not track commit "${commitHash}" to a pull request, it may have been committed directly to the branch`)
-    return null
-  }
-  const title = commitInfo.split('title="')[1].split('"')[0]
-  if (!title.startsWith(TITLE_PREFIX)) {
-    console.warn(`WARNING: Unknown PR title for commit "${commitHash}" in PR "${prIdent}"`)
-    return null
+  if (!semver.valid(current)) {
+    // since we don't seem to be on a stabilization branch right now,
+    // pick a placeholder name that will yield the newest branch
+    // as a comparison point.
+    current = 'v999.999.999'
```

Comment:
![It's over 900!](https://vignette.wikia.nocookie.net/dragonball/images/4/4b/VegetaItsOver9000-02.png/revision/latest?cb=20100724145819)

Reply:
> The year is 2348 and Electron v1000 was just released, suddenly release notes stop working, the world grinds to a halt as we wonder what has changed